src/renderer/components/dock/dock-tabs.tsx

```diff
@@ -39,14 +99,54 @@ export const DockTabs = ({ tabs, autoFocus, selectedTab, onChangeTab }: Props) =
     }
   };
 
+
+  useEffect(() => {
+    // update values in store on scroll
+    elem.current.addEventListener("scroll", updateScrollPosition);
+
+    // update current values on resize to show/hide scroll
+    window.addEventListener("resize", onWindowResize);
+
+    // update scroll state if tabs numbers has changed
+    const tabsNumberChangeDisposer = reaction(() => dockStore.tabsNumber, updateStateValues, { fireImmediately: true });
```

Comment:
This doesn't need to be from `tabsNumber`. It could just be a `useEffect` that is dependent on `tabs.length`

Reply:
That way you don't need to make `dockStore` a prop for this component.

Reply:
Makes sense. Changing